### ISSUE TITLE
CAS Nextcloud redirect not working

### ISSUE THREAD

**Issue by mrvcyln** [felixrupp/user_cas#61]
i want to login with cas via nextcloud. I activated the CAS user and group backend plugin. Then I enter CAS Server Hostname, port, path and service url fields from Cas server field. After that, cas login screen comes up and directs me to cas screen. After a user attempt, the next login screen is normally required, and the cas login screen is successfully displayed. How do I redirect?

Screenshots and log file are attached.

![Screen Shot 2019-07-23 at 10 32 34](https://user-images.githubusercontent.com/11958778/61692286-04b01a80-ad36-11e9-8e9a-7dc4765ebdc4.png)
![Screen Shot 2019-07-23 at 10 33 24](https://user-images.githubusercontent.com/11958778/61692297-0974ce80-ad36-11e9-927a-7b6c60286a61.png)
![Screen Shot 2019-07-23 at 10 33 32](https://user-images.githubusercontent.com/11958778/61692312-11347300-ad36-11e9-9e0b-e8f84b2c90de.png)
<img width="853" alt="Screen Shot 2019-07-23 at 10 36 42" src="https://user-images.githubusercontent.com/11958778/61692317-1396cd00-ad36-11e9-8c30-619682a45d5b.png">
[phpcas.log](https://github.com/felixrupp/user_cas/files/3420780/phpcas.log)

**Comment by felixrupp** [comment]
Hi @mrvcyln 

as described in the readme, the "Service URL" is the URL to your ownCloud/Nextcloud (the "Service" to authenticate the user for, in terms of CAS). Please leave this empty, if you are in an environment without a proxy-server. Only fill this field if you have to use a proxy or you are using a load-balancer configuration, or similar.

Here is the explanation from the readme:

> Service URL: Service URL pointing to your plattform used for CAS authentication and redirection. Useful when behind a reverse proxy. This url must end in /apps/user_cas/login.

Regards,
Felix

**Comment by mrvcyln** [comment]
thanks Felix,
I was able to login this way. But I have this problem. nextcloud directs me to the cas screen when I enter the url. I want to see two login screens as normal login and cas. What should I do about it?

My other question is that I login with users in ldap, but these users are not admin. I want some of them to be admin. How can I arrange this.

Thanks again,

**Comment by felixrupp** [comment]
@mrvcyln If you want to see the regular login screen and want to be able to select between regular Nextcloud login and CAS login, you have to disable: „Force user login using CAS“. If the option is enabled, it forces the user to log in via CAS and redirects immediately.

Regards,
Felix

**Comment by mrvcyln** [comment]
Hi Felix,

I have another problem. I'm doing cas nextcloud openldap authentication. I'm trying to login from the cas login screen in nextcloud with the user I added with Openldap and password. But I get an error. When I try to login from cas login screen with the username and password I created in openldap, I get the error in the screen image. I am sending the phpcas.log file as an attachment.

Thanks,
<img width="1432" alt="Screen Shot 2019-08-21 at 12 34 14" src="https://user-images.githubusercontent.com/11958778/63420994-7bf9cc80-c410-11e9-931e-3504d4b48685.png">
[phpcas.log](https://github.com/felixrupp/user_cas/files/3524627/phpcas.log)
